src/btree/bt_walk.c

```diff
@@ -470,6 +474,10 @@ __tree_walk_internal(WT_SESSION_IMPL *session, WT_REF **refp, uint64_t *walkcntp
                 /* Success, so "couple" has been released. */
                 couple = NULL;
 
+                /* Check if pre-fetch would help. */
+                if (__wt_session_prefetch_check(session, ref))
+                    WT_ERR(__wt_btree_prefetch(session, ref));
```

Comment:
I understand where you're coming from - regarding that prefetch is ultimately an optimisation, but I do think it's worth returning errors when prefetch is turned on, e.g. especially during validation. I'm open to opinions on this though.

Reply:
That's a reasonable take. Swallowing errors for an optimisation is something we do in the rest of the code though - for example, in the POSIX filesystem code, we try to mmap in files, but if it fails we swallow it and mark the handle as non-mappable. Taking down the entire server because some optional functionality failed seems excessive to me.

To move this forward, I'd be happy to file a ticket and discuss it there instead of having this conversation block the PR.

Reply:
@korteland Yes please, let's file a new ticket for this.

Reply:
Made WT-11759.

Reply:
Redundant comment is still there.